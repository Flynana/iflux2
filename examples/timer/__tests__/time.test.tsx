import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { StoreProvider } from 'iflux2'
import Timer from '../src/component/timer'
import Store from '../src/store'

@StoreProvider(Store)
class TimerApp extends React.Component<any, any> {
  render() {
    return <Timer />
  }
}

describe('Relax Counter', () => {
  it('test counter', () => {
    const component = renderer.create(<TimerApp />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    window['store'].dispatch('start');
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})