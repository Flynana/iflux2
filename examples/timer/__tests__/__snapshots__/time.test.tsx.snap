exports[`Relax Counter test counter 1`] = `
<div>
  <a
    href="javascript:void(0);"
    onClick={[Function]}>
    start
  </a>
  <span
    style={
      Object {
        "color": "red",
        "fontSize": 18,
        "marginLeft": 10,
        "marginRight": 10,
      }
    }>
    0
  </span>
  <a
    href="javascript:void(0);"
    onClick={[Function]}>
    reset
  </a>
</div>
`;

exports[`Relax Counter test counter 2`] = `
<div>
  <a
    href="javascript:void(0);"
    onClick={[Function]}>
    start
  </a>
  <span
    style={
      Object {
        "color": "red",
        "fontSize": 18,
        "marginLeft": 10,
        "marginRight": 10,
      }
    }>
    1
  </span>
  <a
    href="javascript:void(0);"
    onClick={[Function]}>
    reset
  </a>
</div>
`;
